dbms/src/Storages/System/StorageSystemParts.cpp

```diff
@@ -102,4 +107,76 @@ void StorageSystemParts::processNextStorage(MutableColumns & columns, const Stor
     }
 }
 
+/**
+  * Implements system table 'detached_parts' which allows to get information
+  * about detached data parts for tables of MergeTree family.
+  * We don't use StorageSystemPartsBase, because it introduces virtual _state
+  * column and column aliases which we don't need.
+  */
+class StorageSystemDetachedParts :
+    public ext::shared_ptr_helper<StorageSystemDetachedParts>,
+    public IStorage
+{
+public:
+    std::string getName() const override { return "SystemDetachedParts"; }
+    std::string getTableName() const override { return "detached_parts"; }
+
+protected:
+    explicit StorageSystemDetachedParts()
+    {
+        setColumns(ColumnsDescription{{
+            {"database", std::make_shared<DataTypeString>()},
+            {"table", std::make_shared<DataTypeString>()},
+            {"partition", std::make_shared<DataTypeString>()},
+            {"name", std::make_shared<DataTypeString>()},
+            {"reason", std::make_shared<DataTypeString>()},
+            {"min_block", std::make_shared<DataTypeInt64>()},
```

Comment:
Let all corresponding fields be named and typed **exactly** as in system.parts.

Reply:
Renamed to partition_id, min_block_number and max_block_number.